src/core/analysis-context.ts

```diff
@@ -483,19 +485,26 @@ export class AnalysisContext {
     }
   }
 
-  /**
-   * Returns true if the url given is resovable by the Analyzer's `UrlResolver`.
-   */
-  canResolveUrl(url: PackageRelativeUrl): boolean {
-    return this.resolver.canResolve(url);
-  }
-
   /**
    * Resolves a URL with this Analyzer's `UrlResolver` or returns the given
    * URL if it can not be resolved.
    */
-  resolveUrl(url: PackageRelativeUrl): ResolvedUrl {
-    return this.resolver.canResolve(url) ? this.resolver.resolve(url) :
-                                           (url as any as ResolvedUrl);
+  resolveUrl(url: PackageRelativeUrl): ResolvedUrl|undefined {
+    const resolved = this.resolver.resolve(url);
+    if (resolved === undefined) {
+      return url as any as ResolvedUrl;
+    }
+    return resolved;
+  }
+
+  resolveUrls(urls: PackageRelativeUrl[]): ResolvedUrl[] {
+    const results = [];
```

Comment:
I do like functional style for this sort of this: `return urls.map((u) => this.resolveUrl(u)).filter((u) => u !== undefined)`